src/network.cpp

```diff
@@ -388,3 +388,97 @@ void Network::dumpNetwork() const
     LOG_DEBUG("Type: {}", magic_enum::enum_name(m_type));
 }
 #endif
+
+#ifdef BUILD_TESTING
+#include <cloysterhpc/tests.h>
+
+TEST_SUITE("Network class")
+{
+    TEST_CASE("Test Network setters and getters")
+    {
+        auto testNetwork = Network(Network::Profile::External);
+
+        SUBCASE("Get m_profile")
+        {
+            CHECK(testNetwork.getProfile() == Network::Profile::External);
+        }
+
+        SUBCASE("Set and get m_address")
+        {
+            boost::asio::ip::address address
+                = boost::asio::ip::make_address("192.168.0.1");
+            testNetwork.setAddress(address);
+            CHECK(testNetwork.getAddress() == address);
+        }
+
+        SUBCASE("Set and get m_subnetMask")
+        {
+            boost::asio::ip::address subnetMask
+                = boost::asio::ip::make_address("255.255.255.252");
+            testNetwork.setSubnetMask(subnetMask);
+            CHECK(testNetwork.getSubnetMask() == subnetMask);
+        }
+
+        SUBCASE("Set and get m_gateway")
+        {
+            boost::asio::ip::address gateway
+                = boost::asio::ip::make_address("0.0.0.0");
+            testNetwork.setGateway(gateway);
+            CHECK(testNetwork.getGateway() == gateway);
+        }
+
+        SUBCASE("Set and get m_vlan")
+        {
+            testNetwork.setVLAN(2000);
+            CHECK(testNetwork.getVLAN() == 2000);
+        }
+
+        SUBCASE("Set and get m_domainName")
+        {
+            std::string domainName = "cloyster.com";
+            testNetwork.setDomainName(domainName);
+            CHECK(testNetwork.getDomainName() == domainName);
+        }
+
+        SUBCASE("Set and get m_nameservers")
+        {
+            //@TODO must fix the setNameservers function
+        }
+    }
+
+    TEST_CASE("Test Network fetchers")
+    {
+
+        Network testNetwork;
+        auto testInterface = "enp0s25";
```

Comment:
Does this is a mock or just a fixed interface name from the OS you're running? If we change this to `enp1s0f0d0` for instance, it will continue to pass the tests? If not we have a problem here.